src/parser/name_resolution.c

```diff
@@ -2633,8 +2869,10 @@ pt_bind_names (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *continue
       pt_bind_scope (parser, bind_arg);
 
       (void) pt_resolve_hint (parser, node);
-
-      parser_walk_leaves (parser, node, pt_bind_names, bind_arg, pt_bind_names_post, bind_arg);
+      if (node->info.update.spec->info.spec.remote_server_name == NULL)
```

Comment:
Why doesn't db_link do it?

Reply:
remote DML (update/delete/insert/merge) is handled at server-side and the DML query is rewriten at pt_rewrite_dblink.